src/cursor/cur_hs.c

```diff
@@ -279,7 +323,7 @@ __curhs_prev_visible(WT_SESSION_IMPL *session, WT_CURSOR_HS *hs_cursor)
 
     WT_ERR(__wt_scr_alloc(session, 0, &datastore_key));
 
-    for (; ret == 0; ret = __wt_hs_cursor_prev(session, file_cursor)) {
```

Comment:
We shouldn't replace this function with std cursor call. We need to use the low level function to ensure we see everything in history store.

Reply:
The function of prev and next is to call the low level function __wt_hs_cursor_prev  and check the visibility using __curhs_prev_visible and __curhs_next_visible until it finds the next visible record to user.